api/src/main/java/ai/djl/ndarray/BaseNDManager.java

```diff
@@ -62,7 +75,12 @@ public Device getDevice() {
     @Override
     public String toString() {
         String parentUID = parent == null ? "No Parent" : ((BaseNDManager) parent).uid;
-        return "UID: "
+        String parentName = parent == null ? "" : parent.getName();
```

Comment:
if there is name, only print name without uid, otherwise use UID

Reply:
There could be cases the manager iteself have name but parent not